sql/plan/show_indexes.go

```diff
@@ -67,60 +67,118 @@ func (n *ShowIndexes) Children() []sql.Node { return nil }
 // RowIter implements the Node interface.
 func (n *ShowIndexes) RowIter(*sql.Context) (sql.RowIter, error) {
 	return &showIndexesIter{
-		db:       n.Database.Name(),
+		db:       n.Database,
 		table:    n.Table,
 		registry: n.Registry,
 	}, nil
 }
 
 type showIndexesIter struct {
-	db       string
+	db       sql.Database
 	table    string
 	registry *sql.IndexRegistry
-	indexes  []sql.Index
-	pos      int
+
+	idxs *indexesToShow
 }
 
 func (i *showIndexesIter) Next() (sql.Row, error) {
 	if i.registry == nil {
 		return nil, io.EOF
 	}
 
-	if i.indexes == nil {
-		i.indexes = i.registry.IndexesByTable(i.db, i.table)
+	if i.idxs == nil {
+		i.idxs = &indexesToShow{
+			indexes: i.registry.IndexesByTable(i.db.Name(), i.table),
+		}
 	}
 
-	if i.pos >= len(i.indexes) {
-		i.Close()
-		return nil, io.EOF
+	show, err := i.idxs.next()
+	if err != nil {
+		return nil, err
 	}
 
-	idx := i.indexes[i.pos]
+	var nullable string
+	columnName, expression := "NULL", show.expression
+	if ok, null := isColumn(show.expression, i.db.Tables()[i.table]); ok {
+		columnName, expression = expression, columnName
+		if null {
+			nullable = "YES"
+		}
+	}
 
-	i.pos++
 	return sql.NewRow(
-		i.table,      // "Table" string
-		int32(1),     // "Non_unique" int32
-		idx.ID(),     // "Key_name" string
-		int32(0),     // "Seq_in_index" int32
-		"NULL",       // "Column_name" string
-		"",           // "Collation" string
-		int64(0),     // "Cardinality" int64
-		int64(0),     // "Sub_part" int64
-		"",           // "Packed" string
-		"",           // "Null" sting
-		idx.Driver(), // "Index_type" string
-		"",           // "Comment" string
-		"",           // "Index_comment" string
-		"YES",        // "Visible" string
-		"NULL",       // "Expression" string
+		i.table,             // "Table" string
+		int32(1),            // "Non_unique" int32, Values [0, 1]
+		show.index.ID(),     // "Key_name" string
+		show.exPosition+1,   // "Seq_in_index" int32
+		columnName,          // "Column_name" string
+		"NULL",              // "Collation" string, Values [A, D, NULL]
+		int64(0),            // "Cardinality" int64 (returning 0, it is not being calculated for the moment)
+		"NULL",              // "Sub_part" int64
```

Comment:
"NULL" should be `nil`, right? or are there writing the NULL string?

Reply:
The are writing the `NULL` string

Reply:
LGTM then.